src/Symfony/Component/Notifier/Bridge/GatewayApi/Tests/GatewayApiTransportTest.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Symfony\Component\Notifier\Bridge\GatewayApi\Tests;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\HttpClient\MockHttpClient;
+use Symfony\Component\Notifier\Bridge\GatewayApi\GatewayApiTransport;
+use Symfony\Component\Notifier\Message\ChatMessage;
+use Symfony\Component\Notifier\Message\SmsMessage;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+
+final class GatewayApiTransportTest extends TestCase
+{
+    public function testSend(): void
+    {
+        $response = $this->createMock(ResponseInterface::class);
+        $response->expects($this->exactly(2))
+            ->method('getStatusCode')
+            ->willReturn(200);
+
+        $client = new MockHttpClient(static function () use ($response): ResponseInterface {
+            return $response;
+        });
+
+        $transport = $this->getTransport($client);
+        $message = new SmsMessage('3333333333', 'Test Messgage');
+        $sentMessage = $transport->send($message);
+        $this->assertNotNull($sentMessage);
+    }
+
+    public function testSupportsSmsMessage(): void
+    {
+        $transport = $this->getTransport();
+        $message = new SmsMessage('3333333333', 'Test Messgage');
+        $this->assertTrue($transport->supports($message));
+    }
+
+    public function testNotSupportsChatMessage(): void
+    {
+        $transport = $this->getTransport();
+        $message = new ChatMessage('3333333333');
+        $this->assertFalse($transport->supports($message));
+    }
+
+    private function getTransport(MockHttpClient $client = null): GatewayApiTransport
+    {
+        if (null == $client) {
+            $this->createMock(ResponseInterface::class);
```

Comment:
Looks like this is not used 🤔

Reply:
Yes, it is used 3 times.

![image](https://user-images.githubusercontent.com/3647884/101811039-d5ac6b80-3b19-11eb-96e2-b2c974c77a3a.png)

Reply:
Hmm but this mock is neither returned nor assigned to any variable 🧐

Reply:
Sorry, you are right. I was thinking you were talking about `getTransport`

Reply:
Thanks